pkg/controller/servicebinding/servicebinder.go

```diff
@@ -298,7 +298,37 @@ func (b *serviceBinder) bind() (reconcile.Result, error) {
 
 	b.logger.Debug("Saving data on intermediary secret...")
 
-	secretObj, err := b.secret.createOrUpdate(b.envVars, b.sbr.AsOwnerReference())
+	if b.sbr.Status.Secret != "" {
+
+		// fetch SBR secret
+		b.logger.WithValues("Namespace", b.secret.ns, "Name", b.secret.name)
+		resourceClient := b.secret.buildResourceClient()
+		existingSecret, err := resourceClient.Get(
+			b.sbr.Status.Secret,
+			metav1.GetOptions{},
+		)
+		if err != nil {
+			b.logger.Error(err, "Error fetching secret")
+			return requeueError(err)
+		}
+		existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
+
+		// payload secret
+		payload := make(map[string]interface{})
+		for k, v := range b.secret.data {
+			payload[k] = base64.StdEncoding.EncodeToString(v)
+		}
+
+		if !nestedMapComparison(existingSecretData, payload).Success {
+			b.logger.Info("Secret is different")
+			if err := b.binder.unbind(); err != nil && err != errEmptyApplication {
```

Comment:
Why are we calling `unbind` here? It wasn't being called in the original implementation.

Reply:
If we don't call unbind then the deployment still has the old secret reference.

We remove the old secret reference and then add the new one.

Reply:
Then `bind` isn't doing its job correctly: its work is to fix the secret name in the PodSpec.

I expect `bind` to update that field without calling `unbind` in the middle of its logic. Probably having the previous secret name is enough to remove it from the list before adding the new one without such expensive call.

Reply:
Done